src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraConnectorITCase.java

```diff
@@ -0,0 +1,912 @@
+            "INSERT INTO "
+                    + KEYSPACE
+                    + "."
+                    + TABLE_NAME_VARIABLE
+                    + " ("
+                    + TUPLE_ID_FIELD
+                    + ", "
+                    + TUPLE_COUNTER_FIELD
+                    + ", "
+                    + TUPLE_BATCHID_FIELD
+                    + ") VALUES (?, ?, ?)";
+    private static final String SELECT_DATA_QUERY =
+            "SELECT * FROM " + KEYSPACE + "." + TABLE_NAME_VARIABLE + ';';
+
+    private static final Random random = new Random();
+    private int tableID;
+
+    private static final ArrayList<Tuple3<String, Integer, Integer>> collection =
+            new ArrayList<>(20);
+    private static final ArrayList<Row> rowCollection = new ArrayList<>(20);
+
+    private static final TypeInformation[] FIELD_TYPES = {
+        BasicTypeInfo.STRING_TYPE_INFO, BasicTypeInfo.INT_TYPE_INFO, BasicTypeInfo.INT_TYPE_INFO
+    };
+
+    static {
+        for (int i = 0; i < 20; i++) {
+            collection.add(new Tuple3<>(UUID.randomUUID().toString(), i, 0));
+            rowCollection.add(Row.of(UUID.randomUUID().toString(), i, 0));
+        }
+    }
+
+    private static Class<? extends Pojo> annotatePojoWithTable(String keyspace, String tableName) {
+        return new ByteBuddy()
+                .redefine(Pojo.class)
+                .name("org.apache.flink.streaming.connectors.cassandra.Pojo" + tableName)
+                .annotateType(createTableAnnotation(keyspace, tableName))
+                .make()
+                .load(Pojo.class.getClassLoader())
+                .getLoaded();
+    }
+
+    @NotNull
+    private static Table createTableAnnotation(String keyspace, String tableName) {
+        return new Table() {
+
+            @Override
+            public String keyspace() {
+                return keyspace;
+            }
+
+            @Override
+            public String name() {
+                return tableName;
+            }
+
+            @Override
+            public boolean caseSensitiveKeyspace() {
+                return false;
+            }
+
+            @Override
+            public boolean caseSensitiveTable() {
+                return false;
+            }
+
+            @Override
+            public String writeConsistency() {
+                return "";
+            }
+
+            @Override
+            public String readConsistency() {
+                return "";
+            }
+
+            @Override
+            public Class<? extends Annotation> annotationType() {
+                return Table.class;
+            }
+        };
+    }
+
+    // ------------------------------------------------------------------------
+    //  Utility methods
+    // ------------------------------------------------------------------------
+
+    public static CassandraContainer createCassandraContainer() {
+        CassandraContainer cassandra = new CassandraContainer(DockerImageVersions.CASSANDRA_4_0);
```

Comment:
Don't rely on the Flink DockerImageVersions; this entry will be removed eventually.

Reply:
> We'd like to preserve the git history; can you integrate the `cas_rewrite` branch from this repo into your PR?

That makes total sense. I'll rebase on the branch. IMHO it would be interesting to add an item to your migration doc to inform contributors to keep history, that way you would not need to create the branch yourself.

Reply:
:+1: